### DISCUSSION TITLE
Export values with M2O selections, fields sorting and exports

### DISCUSSION THREAD

**Discussion by adrchen** [directus/directus#5043] [Archive]
Running **v.9.0.0-rc.57**, is it possible to change the default setting of saving into the database the id of the M2O entry value into:
- The label used? For example, if the label is `first name last name`, then `first name last name` being saved instead of 1 (which would be the entry id);
- Another entry related to that data? For example, if the label is `identification number - first name last name`, then save to the database `first name last name`.

Also, regardless of the order saved while configuring a Collection, once displayed the fields are ordered alphabetically. This can be fixed by manually ordering it again, but when it's exported, the columns are ordered alphabetically as well. Any way to export it in the same order as displayed?

**Comment by rijkvanzanten** [comment]
> is it possible to change the default setting of saving into the database the id of the M2O entry value into:

No this isn't possible. I don't think it's something we'll be looking to support either, as it inherently goes against what foreign keys are in a relational database 🤔 

> Any way to export it in the same order as displayed?

Am I right in understanding that you're asking to have the `export` query param have the order of the fields match the order as configured in settings?

**Reply by adrchen** [reply]
> Am I right in understanding that you're asking to have the `export` query param have the order of the fields match the order as configured in settings?

Yes. I'm more like asking if it's currently possible.

**Reply by rijkvanzanten** [reply]
The export query param will export whatever the regular API response is to a file, so that's currently not sorted 👍🏻

**Reply by adrchen** [reply]
> The export query param will export whatever the regular API response is to a file, so that's currently not sorted 👍🏻

Thanks for the prompt reply! How can I sort the regular API response to a file?

**Reply by rijkvanzanten** [reply]
The `sort` parameter only sorts the output rows of the data you fetch. There is no way to sort the actual fields per row themselves
